staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -594,10 +600,28 @@ func patchResource(
 				}
 				originalObjJS, originalPatchedObjJS = originalJS, patchedJS
 			case types.StrategicMergePatchType:
-				originalMap, patchMap, err := strategicPatchObject(codec, currentObject, patchJS, objToUpdate, versionedObj)
+				// Since the patch is applied on versioned objects, we need to convert the
+				// current object to versioned representation first.
+				gvks, _, err := typer.ObjectKinds(versionedObj)
 				if err != nil {
 					return nil, err
 				}
+				gvk := gvks[0]
+				currentVersionedObject, err := convertor.ConvertToVersion(currentObject, gvk.GroupVersion())
```

Comment:
I don't understand how to pass gvks to it - it's a different type (it just doesn't compile for me).

Regarding UnsafeConvert - yes, this object is throwaway, but to do it:
- I would need access to runtime.Scheme (which I don't have now) or extend Convertor interface (both are not super easy)
- it shouldn't be worse that what we have now - see my comment below).

So I just dropped a TODO for it. I can work on it in a subsequent PR (though it probably won't hit 1.6).